R/functions-Spectrum.R

```diff
@@ -526,19 +526,20 @@ validSpectrum <- function(object) {
 #'
 #' @noRd
 .spectrum_header <- function(x) {
-    res <- data.frame(acquisitionNum = acquisitionNum(x),
+    res <- data.frame(acquisitionNum = if (length(x@acquisitionNum)) x@acquisitionNum
+                                       else NA_integer_,
                       msLevel = msLevel(x),
-                      polarity = ifelse(length(x@polarity), x@polarity, NA_integer_),
+                      polarity = if (length(x@polarity)) x@polarity else NA_integer_,
```

Comment:
:+1: `if ... else ...` is much better (and faster) than `ifelse` if the testing condition isn't vectorised.